### PR TITLE
@craigspaeth Adds contributing authors

### PR DESCRIPTION
**Pull request by kanaabe**
Also adds a `all_by_author` query param which finds articles that an author has been the main author AND contributing author of. Will use in Force for adding articles to a contributor's profile page. 

contributing

### CODE REVIEW

File: api/apps/articles/model.coffee [hunk 1]
```diff
@@ -140,6 +142,12 @@ toQuery = (input, callback) ->
     query.auction_id = ObjectId input.auction_id if input.auction_id
     query.vertical_id = ObjectId input.vertical_id if input.vertical_id
     query.biography_for_artist_id = ObjectId input.biography_for_artist_id if input.biography_for_artist_id
+
+    query.$or = [
+      { author_id: ObjectId(input.all_by_author) }
+      { contributing_authors: ObjectId(input.all_by_author) }
```
Comment (craigspaeth) [inline api/apps/articles/model.coffee]: Niice!

File: client/apps/edit/components/admin/index.coffee [hunk 2]
```diff
@@ -145,6 +147,36 @@ module.exports = class EditAdmin extends Backbone.View
...
+      placeholder: 'Search by user name or email...'
+      filter: (users) -> for user in users
+        { id: user.id, value: _.compact([user.name, user.email]).join(', ') }
+      selected: (e, item, items) =>
+        @article.save contributing_authors: _.pluck items, 'id'
+      removed: (e, item, items) =>
+        @article.save contributing_authors: _.without(_.pluck(items, 'id'),item.id)
+    if ids = @article.get 'contributing_authors'
+      @authors = []
+      async.each ids, (id, cb) =>
```
Comment (craigspaeth) [inline client/apps/edit/components/admin/index.coffee]: Nice! Looks like you grasped async pretty easily! Let me know if anything is unclear with how that control flow lib works.

**Comment by craigspaeth** [comment]
Solid work! Code is great!

**Comment by kanaabe** [comment]
@craigspaeth Do you think we should also be saving user `name` and profile `hrefs` ? I realize we'll need to fetch users on the Force/MG to show the bylines. Not sure what is cleaner/best practice here for API design.

**Comment by craigspaeth** [comment]
@kanaabe Denormalizing some user data into the articles e.g. `article.contributing_authors: [{ id: ObjectId(), name: '', profile_id: '', profile_handle: '' }]` is probably a good idea. We already do this for the primary author [see here](https://github.com/artsy/positron/blob/master/api/apps/articles/model.coffee#L224). So :+1: from me. Either way if it's just an array of ids the field should probably be renamed to `contributing_author_ids` to be specific.

**Pull request merged**